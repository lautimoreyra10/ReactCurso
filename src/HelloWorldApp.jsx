// function App () {
//     //document.createElement...
//     return (<h1>Hola Mundo</h1>);
// }


// export default App;

// Esta es otra forma de hacerlo, utilizando el shortcut "rafc"
export const HelloWorldApp = () => {
    
  return (
    <h1>Hello World App</h1>
  );
}

//Otra forma mas
//export const HelloWorldApp = () => (<h1>Arrow Function</h1>)
