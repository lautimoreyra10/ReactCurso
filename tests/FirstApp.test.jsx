/** @jest-environment jsdom */
import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en FirstApp', () => { 
    
    test('Debe hacer match con snapshot', () => { 
        const title = 'Hola, soy yo';
        const { container } = render( <FirstApp title={ title } /> )
        
        expect(container).toMatchSnapshot();
        });
    test('Debe mostrar el tÃ­tulo en un h1', () => { 
        const title = 'Hola, soy yo';
        const { container, getByText } = render( <FirstApp title={ title } /> )

        expect(getByText(title) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title );
     })
 });