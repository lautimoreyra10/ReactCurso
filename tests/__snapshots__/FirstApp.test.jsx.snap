// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con snapshot 1`] = `
<div>
  <h1>
    Hola, soy yo
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Lautaro Moreyra
  </p>
</div>
`;
